### PR TITLE
Decoder

### LINKED ISSUE #8: Implement Decoder
- [x] Simple decoding (w/out codegen yet)
 - [x] Instruction class (OpType, Reg1, Reg2, Imm...)

### LINKED ISSUE #18: Turn off formatting on pipeline
Add flag to disable formatting, because we don't need to format project in github actions

### PR DESCRIPTION
**Pull request by derzhavin3016**
- Renamed Reg to RegVal
- Added Addr & RegId types
- Added kRegNum constant
- Added enum w/ operations

closing #8

### CODE REVIEW

File: include/decoder/decoder.hh [hunk 1]
```diff
@@ -3,12 +3,55 @@
 
 #include "common/common.hh"
 #include "common/inst.hh"
+#include <type_traits>
```
Comment (Tako-San) [inline include/decoder/decoder.hh]: nit: Move it up, separately from our include files

Comment (derzhavin3016) [reply include/decoder/decoder.hh]: Fixed

File: include/common/common.hh [hunk 2]
```diff
@@ -6,7 +6,58 @@
...
+
+  if constexpr (newSize == oldSize)
+    return word;
+
+  Word zeroed = getBits<oldSize - 1, 0>(word);
+  constexpr Word mask = Word(1) << (oldSize - 1);
+  Word res = (zeroed ^ mask) - mask;
+
+  return getBits<newSize - 1, 0>(res);
+}
```
Comment (Tako-San) [inline include/common/common.hh:104]: nit: Maybe it is good idea to provide a nice-looking comment with example of work of this algorithm

Comment (derzhavin3016) [reply include/common/common.hh:104]: изображение

File: src/decoder/CMakeLists.txt [hunk 3]
```diff
@@ -1 +1,49 @@
-add_library(decoder decoder.cc)
+find_package(
+  Python3
+  COMPONENTS Interpreter
+  REQUIRED)
+
+set(SIM_RISCV_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../thirdparty/riscv-opcodes)
```
Comment (Tako-San) [inline src/decoder/CMakeLists.txt:6]: nit: maybe it is better to use CMAKE_SOURCE_DIR

File: include/common/common.hh (hunk 2 above)
Comment (Tako-San) [reply include/common/common.hh:104]: Cool

**Review by Tako-San — APPROVED** [review]
LGTM

**Pull request merged**